File: 2766-find-the-prefix-common-array-of-two-arrays/2766-find-the-prefix-common-array-of-two-arrays.js
/**
 * @param {number[]} A
 * @param {number[]} B
 * @return {number[]}
 */
var findThePrefixCommonArray = function(A, B) {
    return findThePrefixCommonArrays(A,B);
};
function findThePrefixCommonArrays(a, b) {
  const map1 = new Map();
  const map2 = new Map();
  var count = 0;
  return a.map((val, index) => {
    map1.set(a[index], 1);
    map2.set(b[index], 1);
    if (map1.has(b[index]) && map2.has(a[index])) {
      if (a[index] === b[index]) count += 1;
      else count += 2;
      return count;
    } else if (map1.has(b[index])) {
      count += 1;
      return count;
    } else if (map2.has(a[index])) {
      count += 1;
      return count;
    } else return count;
  });
}

File: 2766-find-the-prefix-common-array-of-two-arrays/2766-find-the-prefix-common-array-of-two-arrays.java
class Solution {
    public int[] findThePrefixCommonArray(int[] A, int[] B) {
        return getCarrUsingStream(A,B);
    }
 public  int[] getCarrUsingStream(int[] a, int[] b) {
        Map<Integer, Integer> map1 = new HashMap<>();
        Map<Integer, Integer> map2 = new HashMap<>();
        int[] count = {0};
       return  IntStream.range(0,a.length)
                .map(index ->{
                    map1.put(a[index],1);
                    map2.put(b[index],1);
                    if (map1.containsKey(b[index]) && map2.containsKey(a[index])){
                        if (a[index] == b[index])
                            count[0] += 1;
                        else
                            count[0] += 2;
                        return count[0];
                    } else if (map1.containsKey(b[index])) {
                        count[0] += 1;
                        return count[0];
                    } else if (map2.containsKey(a[index])) {
                        count[0] += 1;
                        return count[0];
                    }
                    return count[0];
                }).toArray();
    }
}